sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/OpenTelemetrySdkAutoConfiguration.java

```diff
@@ -0,0 +1,351 @@
+        case "prometheus":
+          if (metricsConfigured) {
+            throw new IllegalStateException(
+                "Multiple metrics exporters configured. Only one metrics exporter can be "
+                    + "configured at a time.");
+          }
+          configurePrometheusMetrics(config, meterProvider);
+          metricsConfigured = true;
+          break;
+        default:
+          // Ignore
+      }
+    }
+
+    SdkTracerProviderBuilder tracerProviderBuilder =
+        SdkTracerProvider.builder()
+            .setResource(resource)
+            .setTraceConfig(configureTraceConfig(config));
+
+    List<SpanExporter> spanExporters = new ArrayList<>();
+    for (String exporterName : exporterNames) {
+      exporterName = exporterName.toLowerCase(Locale.ROOT);
+      switch (exporterName) {
+        case "otlp":
+        case "otlp_span":
+          spanExporters.add(configureOtlpSpans(config));
+          break;
+        case "jaeger":
+          spanExporters.add(configureJaeger(config));
+          break;
+        case "jaeger-thrift":
+          spanExporters.add(configureJaegerThrift(config));
+          break;
+        case "zipkin":
+          spanExporters.add(configureZipkin(config));
+          break;
+        case "logging":
+          spanExporters.add(new LoggingSpanExporter());
+          break;
+        default:
+          // Ignore
+      }
+    }
+
+    if (!spanExporters.isEmpty()) {
+      tracerProviderBuilder.addSpanProcessor(configureSpanProcessor(config, spanExporters));
+    }
+
+    for (SdkTracerProviderConfigurer configurer :
+        ServiceLoader.load(SdkTracerProviderConfigurer.class)) {
+      configurer.configure(tracerProviderBuilder);
+    }
+
+    List<TextMapPropagator> propagators = new ArrayList<>();
+    for (String propagatorName : config.getCommaSeparatedValues("otel.propagators")) {
+      switch (propagatorName) {
+        case "tracecontext":
+          propagators.add(W3CTraceContextPropagator.getInstance());
+          break;
+        case "baggage":
+          propagators.add(W3CBaggagePropagator.getInstance());
+          break;
+        case "b3":
+          propagators.add(B3Propagator.getInstance());
+          break;
+        case "b3multi":
+          propagators.add(B3Propagator.builder().injectMultipleHeaders().build());
+          break;
+        case "jaeger":
+          propagators.add(JaegerPropagator.getInstance());
+          break;
+        case "ottracer":
+          propagators.add(OtTracerPropagator.getInstance());
+          break;
+        case "xray":
+          propagators.add(AwsXRayPropagator.getInstance());
+          break;
+        default:
+          // Ignore
+      }
+    }
+
+    return OpenTelemetrySdk.builder()
+        .setTracerProvider(tracerProviderBuilder.build())
+        .setPropagators(ContextPropagators.create(TextMapPropagator.composite(propagators)))
+        .build();
+  }
+
+  private static void configureOtlpMetrics(
```

Comment:
I think I'd break each of these methods out into their own class so it's clear how to add new ones to the auto-config, rather than having to dig into this beast to find them. :)